File: src/Java/myservicebus/src/main/java/com/myservicebus/serialization/RawJsonMessageDeserializer.java
package com.myservicebus.serialization;

import com.fasterxml.jackson.databind.ObjectMapper;
import java.nio.charset.StandardCharsets;
import java.util.Map;

public final class RawJsonMessageDeserializer implements MessageDeserializer {
    private final ObjectMapper mapper;
    private final MessageHeaderConvention headerConvention;

    public RawJsonMessageDeserializer() {
        this(JsonSerializationDefaults.createObjectMapper(), MassTransitHeaderConvention.INSTANCE);
    }

    public RawJsonMessageDeserializer(MessageHeaderConvention headerConvention) {
        this(JsonSerializationDefaults.createObjectMapper(), headerConvention);
    }

    public RawJsonMessageDeserializer(ObjectMapper mapper) {
        this(mapper, MassTransitHeaderConvention.INSTANCE);
    }

    public RawJsonMessageDeserializer(ObjectMapper mapper, MessageHeaderConvention headerConvention) {
        if (mapper == null) {
            throw new IllegalArgumentException("mapper must not be null");
        }
        this.mapper = mapper;
        this.headerConvention = headerConvention;
    }

    @Override
    public String getContentType() {
        return DefaultInboundMessageResolver.RAW_JSON_CONTENT_TYPE;
    }

    @Override
    public InboundMessage deserialize(MessageBody body, Map<String, Object> headers) {
        return new RawJsonInboundMessage(body.getBytes(), headers, mapper, headerConvention);
    }

    @Override
    public MessageBody getMessageBody(String text) {
        return new ByteArrayMessageBody(text.getBytes(StandardCharsets.UTF_8));
    }
}

File: src/MyServiceBus/Serialization/RawJsonMessageDeserializer.cs
using System.Text;
using System.Text.Json;

namespace MyServiceBus.Serialization;

public sealed class RawJsonMessageDeserializer : IMessageDeserializer
{
    private readonly IMessageHeaderConvention _headerConvention;
    private readonly JsonSerializerOptions _jsonSerializerOptions;

    public RawJsonMessageDeserializer()
        : this(JsonSerializationDefaults.CreateOptions(), MassTransitHeaderConvention.Instance)
    {
    }

    public RawJsonMessageDeserializer(IMessageHeaderConvention headerConvention)
        : this(JsonSerializationDefaults.CreateOptions(), headerConvention)
    {
    }

    public RawJsonMessageDeserializer(
        JsonSerializerOptions jsonSerializerOptions,
        IMessageHeaderConvention? headerConvention = null)
    {
        ArgumentNullException.ThrowIfNull(jsonSerializerOptions);
        _jsonSerializerOptions = jsonSerializerOptions;
        _headerConvention = headerConvention ?? MassTransitHeaderConvention.Instance;
    }

    public string ContentType => InboundMessageResolver.RawJsonContentType;

    public IInboundMessage Deserialize(MessageBody body, IDictionary<string, object> headers)
        => new RawJsonMessageContext(body.GetBytes(), headers, _jsonSerializerOptions, _headerConvention);

    public MessageBody GetMessageBody(string text)
        => new ByteArrayMessageBody(Encoding.UTF8.GetBytes(text));
}
